packages/angular/cli/commands/definitions.json

```diff
@@ -30,12 +30,9 @@
       "type": "object",
       "properties": {
         "help": {
-          "type": "boolean",
-          "description": "Shows a help message."
-        },
-        "helpJson": {
-          "type": "boolean",
-          "description": "Shows the metadata associated with each flags, in JSON format."
+          "type": ["boolean", "string"],
+          "description": "Shows a help message. Use '--help=json' as a value to output the help in JSON format.",
+          "default": false
```

Comment:
`"enum": [true, "json"]`?

or more maybe:
```
  "anyOf": [
    {
      "enum": [ true, "json" ]
    },
    {
      "pattern": "^[Jj][Ss][Oo][Nn]$"
    }
  ]
```

Reply:
The first one would not handle the `—no-help` case. The second one wouldn’t list the enumerations properly.

Reply:
the tsconfig schema uses that structure to handle casing and vscode displays the enum values (not sure what its code is doing).

Reply:
I'm ok with it either way though.

Reply:
I would think vscode is showing it from the type itself? The `.d.ts` is cleaner.

I also changed the code of the parser to reject values that are not part of the enum (and adding it to leftovers). I wasn't sure what sort of behaviour would be correct here, TBH. If we set it and it's not part of the enum, then I would consider that a breach of contract. Ignoring the flag might be slightly better but lead to user errors.

I could use some whiteboarding on this actually.